test/model.jl

```diff
@@ -153,6 +153,11 @@ function test_model()
         @test !called
         optimize!(m)
         @test called
+
+        m = Model()
+        JuMP.set_optimize_hook(m, (m ; kwargs=nothing) -> kwargs)
+        @test optimize!(m) === nothing
+        @test optimize!(m, kwargs = 1) == 1
```

Comment:
It might be worth changing the name of `kwargs` for the sake of clarity, e.g., to `my_new_arg`.

Also, a test for the unrecognized keyword argument case?